Source/RunActivity/Viewer3D/Processes/RenderProcess.cs

```diff
@@ -97,10 +99,19 @@ internal RenderProcess(Game game)
             // Set up the rest of the graphics according to the settings.
             GraphicsDeviceManager.SynchronizeWithVerticalRetrace = Game.Settings.VerticalSync;
             GraphicsDeviceManager.PreferredBackBufferFormat = SurfaceFormat.Color;
-            GraphicsDeviceManager.PreferredDepthStencilFormat = DepthFormat.Depth32;
+            GraphicsDeviceManager.PreferredDepthStencilFormat = DepthFormat.Depth24Stencil8;
             GraphicsDeviceManager.IsFullScreen = false;
             GraphicsDeviceManager.PreferMultiSampling = true;
             GraphicsDeviceManager.PreparingDeviceSettings += new EventHandler<PreparingDeviceSettingsEventArgs>(GDM_PreparingDeviceSettings);
+            var screen = Game.Settings.FastFullScreenAltTab ? Screen.FromControl(GameForm) : Screen.PrimaryScreen;
+            if (screen.Primary)
+            {
+                var wa = Screen.PrimaryScreen.WorkingArea;
```

Comment:
Why are we only checking the working area on the primary screen? E.g. Windows 10 has taskbars on every monitor so this seems incorrect.

Reply:
I'd assume this code nor the original code OR/XNA code was built to handle multiple screens, and only handles the primary screen correct. I.e. it fails if screens have different resolution, or taskbars are located differently on each monitor. It may require a rework of the whole functionality.

Reply:
That does not excuse this code. The code is making all kinds of incorrect assumptions and I would like it fixed. For example, the use of working area is only correct when we're in a window (not full screen) and has nothing to do with whether we are on the primary monitor, like the code is checking. We don't need to get multi-monitor support right here but we do need to not screw up the basics like that.